enhancements/installer/multi-disk-setup.md

```diff
@@ -0,0 +1,975 @@
+        Device:         fmt.Sprintf("/dev/disk/by-partlabel/%s", label),
+        Format:         ptr.To(fsType),
+        Label:          ptr.To(label + "part"),
+        MountOptions:   mountOptions,
+        Path:           ptr.To(mountPath),
+        WipeFilesystem: ptr.To(true),
+    })
+
+    // 3. Create systemd mount unit
+    unitName := pathToUnitName(mountPath) + ".mount"
+    config.Systemd.Units = append(config.Systemd.Units, igntypes.Unit{
+        Name:    unitName,
+        Enabled: ptr.To(true),
+        Contents: ptr.To(generateMountUnit(mountPath, label, fsType, mountOptions)),
+    })
+}
+```
+
+For **swap** disks:
+
+```go
+func addSwapSetup(config *igntypes.Config, device string) {
+    label := "swapdisk"
+
+    // 1. Create partition
+    config.Storage.Disks = append(config.Storage.Disks, igntypes.Disk{
+        Device: device,
+        Partitions: []igntypes.Partition{{
+            Label:    ptr.To(label),
+            StartMiB: ptr.To(0),
+            SizeMiB:  ptr.To(0),
+        }},
+        WipeTable: ptr.To(true),
+    })
+
+    // 2. Format as swap
+    config.Storage.Filesystems = append(config.Storage.Filesystems, igntypes.Filesystem{
+        Device:         fmt.Sprintf("/dev/disk/by-partlabel/%s", label),
+        Format:         ptr.To("swap"),
+        WipeFilesystem: ptr.To(true),
+    })
+
+    // 3. Create systemd swap unit
+    config.Systemd.Units = append(config.Systemd.Units, igntypes.Unit{
+        Name:    "dev-disk-by\\x2dpartlabel-swapdisk.swap",
+        Enabled: ptr.To(true),
+        Contents: ptr.To(`
+[Unit]
+Description=Swap on dedicated disk
+
+[Swap]
+What=/dev/disk/by-partlabel/swapdisk
+
+[Install]
+WantedBy=swap.target
+`),
+    })
+}
+```
+
+#### Platform-Specific Disk Mapping
+
+**Azure Example:**
+
+The `platformDiskID` in the `diskSetup` configuration references the `nameSuffix` of an Azure DataDisk. The installer:
+
+1. Matches `diskSetup[i].etcd.platformDiskID` with `dataDisks[i].nameSuffix`
+2. Retrieves the LUN from the corresponding DataDisk
+3. Maps to device path: `/dev/disk/azure/scsi1/lun{LUN}`
+
+```go
+// Azure-specific mapping
+for i, diskSetup := range controlPlane.DiskSetup {
+    if diskSetup.Type == types.Etcd {
+        for _, dataDisk := range controlPlane.Platform.Azure.DataDisks {
+            if diskSetup.Etcd.PlatformDiskID == dataDisk.NameSuffix {
+                device := fmt.Sprintf("/dev/disk/azure/scsi1/lun%d", *dataDisk.Lun)
+                mc, err := ForDiskSetup("master", device, types.Etcd, diskSetup)
+                // Add MachineConfig to manifests
+            }
+        }
+    }
+}
+```
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+This feature can be applied to Hypershift / Hosted Control Planes for worker nodes. The hosted control plane itself runs as pods and does not use this disk setup mechanism. Worker nodes in hosted clusters can be configured with disk setup for swap or user-defined storage needs.
```

Comment:
Not sure on this point. I don't think hosted clusters are created via the installer at all? Are you proposing that the user use installer to generate their nodepool-specific machineconfigs, apply them to new nodepools by hand, and have the MCO reconcile?